rest/specs/src/main/filtered-resources/openapi/ingestapi.yaml

```diff
@@ -0,0 +1,221 @@
+          The client does not have permission.
+        headers:
+          'Content-Version':
+            $ref: '#/components/headers/ContentVersion'
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ErrorMessage'
+    NotImplemented:
+      description: >
+        The requested API version is not supported and therefore not implemented.
+      headers:
+        'Content-Version':
+          $ref: '#/components/headers/ContentVersion'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorMessage'
+    BadRequest:
+      description: >
+        The client message could not understood by the server due to invalid format or syntax.
+      headers:
+        'Content-Version':
+          $ref: '#/components/headers/ContentVersion'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorMessage'
+    DefaultError:
+      description: >
+        Any other possible errors not currently known.
+      headers:
+        'Content-Version':
+          $ref: '#/components/headers/ContentVersion'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorMessage'
+  parameters:
+    Accept-Version:
+      name: Accept-Version
+      in: header
+      description: >
+        The minimal API version that a client using this API will accept
+        responses from.
+      required: true
+      schema:
+        $ref: '#/components/schemas/Version'
+  headers:
+    ContentVersion:
+      description: >
+        The API version used by the server to produce the REST message. The server will accept
+        messages for any minor versions prior to this one.
+      required: true
+      schema:
+        $ref: '#/components/schemas/Version'
+      examples:
+        major.minor:
+          value: '1.2'
+          summary: A released minor version
+        major.minor.patch:
+          value: '1.0.3'
+          summary: A patched version
+        dev:
+          value: '1.2.0-SNAPSHOT'
+          summary: A development version
+  schemas:
+    Product:
+      description: The JSON description of the Product being ingested
+      type: object
+      properties:
+        productInfo:
+          type: object
+          properties:
+            title:
+              type: string
+              description: The title of the Product
+              example: A Picture
+            filename:
+              type: string
+              description: The name of the source file
+              example: product.jpeg
+            bytes:
+              type: integer
+              description: The size of the file payload
+              minimum: 0
+              exclusiveMinimum: true
+              example: 1024
+            mime-type:
+              type: string
+              description: The MIME type of the file being sent
+              example: image/jpeg
```

Comment:
❓ regex for those are free form?

Reply:
I'm not sure what you mean.

Reply:
is there a way to be more specific and get standard validators to pre-validate parameters before it hits our controller? Is so then we should make sure to define the data restrictions as specific as we want it to be. If this is a mime, than I suspect we can set a regex for it (checkout http://regexhero.net/library/6293/mime-type-regex)

Reply:
Ah, yes we can we could pass the request body to a validator before we act on it in the controller. I'll add the regex pattern for it. (thanks for the link)

Reply:
The validation happens automatically as long as one of the spring library is in the classpath